src/EGO Forum Enhancement.ts

```diff
@@ -880,6 +880,160 @@ function addTrashButton(before: HTMLDivElement) {
     before.parentElement?.insertBefore(trashButton, before);
 }
 
+/**
+ * Adds additional buttons to post action bars to make it easier to perform some actions.
+ */
+function addPostActionBarButtons() {
+    const threadId = window.location.href.match(/threads\/(?<post_id>\d+)/);
```

Comment:
I'd recommend using the `data-content-key` attribute in the document's HTML tag to get the ID instead of matching with the URL. This should protect against the URL structure changing. This is already done to get the [forum id](https://github.com/blankdvth/eGOScripts/blob/master/src/EGO%20Forum%20Enhancement.ts#L1113-L1117). Preferably, split getting the thread ID off into a separate function as well (probably under the forum ID one) so that it can be used in the future.